fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProviderPool.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+protocol RuntimeProviderPoolProtocol {
+    func getWalletRuntime(for chain: ChainModel) -> RuntimeProviderProtocol
+    func getFullRuntime(for chain: ChainModel) -> RuntimeProviderProtocol
+}
+
+final class RuntimeProviderPool {
+    let cacheLimit: Int
+    let runtimeProviderFactory: RuntimeProviderFactoryProtocol
+
+    init(cacheLimit: Int, runtimeProviderFactory: RuntimeProviderFactoryProtocol) {
+        self.cacheLimit = cacheLimit
+        self.runtimeProviderFactory = runtimeProviderFactory
+    }
+}
+
+extension RuntimeProviderPool: RuntimeProviderPoolProtocol {
+    func getWalletRuntime(for chain: ChainModel) -> RuntimeProviderProtocol {
+        runtimeProviderFactory.createRuntimeProvider(for: chain)
+    }
+
+    func getFullRuntime(for chain: ChainModel) -> RuntimeProviderProtocol {
+        runtimeProviderFactory.createRuntimeProvider(for: chain)
```

Comment:
implementation is identical to `getWalletRuntime`

Reply:
Yep, marked all unimplemented parts with TODO providing task number